cubeviz/controls/flux_unit_registry.py

```diff
@@ -0,0 +1,289 @@
+
+
+def _add_unit_to_list(unit_list, target_unit):
+    """
+    Given a list of units, add target units
+    after checking for duplicates
+    :param unit_list: list of units or string
+    :param target_unit: unit or string
+    :return: updated unit
+    """
+
+    if isinstance(target_unit, str):
+        current_unit = u.Unit(target_unit)
+        string_unit = target_unit
+    else:
+        current_unit = target_unit
+        string_unit = target_unit.to_string()
+
+    duplicate = False
+    for unit in unit_list:
+        if isinstance(unit, str):
+            unit = u.Unit(unit)
+        if unit == current_unit:
+            duplicate = True
+            break
+    if not duplicate:
+        unit_list.append(string_unit)
+    return unit_list
+
+
+class FluxUnitRegistry:
+    """
+    Saves a list of spectral flux density units.
+    """
+    def __init__(self):
+        self._model_unit = u.Jy  # Will be used for compatibility checks
+        self.runtime_defined_units = []  # Stores list of runtime units
+
+    @staticmethod
+    def _locally_defined_units():
+        """
+        list of defined spectral flux density units.
+        :return: list of unit strings
+        """
+        units = ['Jy', 'mJy', 'uJy',
+                 'W / (m2 Hz)',
+                 'eV / (s m2 Hz)',
+                 'erg / (s cm2)',
+                 'erg / (s cm2 um)',
+                 'erg / (s cm2 Angstrom)',
+                 'erg / (s cm2 Hz)',
+                 'ph / (s cm2 um)',
+                 'ph / (s cm2 Angstrom)',
+                 'ph / (s cm2 Hz)']
+        return units
+
+    def is_compatible(self, unit):
+        """
+        Check if unit can be converted to units
+        in this registry. Uses model unit to check.
+        :param unit: Unit or unit str
+        :return: bool
+        """
+        try:
+            self._model_unit.to(unit, equivalencies=u.spectral_density(3500 * u.AA))
+            return True
+        except u.UnitConversionError:
+            return False
+
+    def get_unit_list(self, current_unit=None):
+        """
+        Returns a list of unit strings in the registry.
+        Adds current_unit if not duplicated. Registered
+        units are added first.
+        :param current_unit: Unit or unit str
+        :return: list of unit str
+        """
+        unit_list = []  # final list
+        item_list = []  # items to be checked if duplicated
+        unit_list.extend(self._locally_defined_units())
+        item_list.extend(self.runtime_defined_units)
+
+        for item in item_list:
+            unit_list = _add_unit_to_list(unit_list, item)
+
+        if current_unit is not None:
+            if isinstance(current_unit, str):
+                current_unit = u.Unit(current_unit)
+            unit_list = _add_unit_to_list(unit_list, current_unit)
+        return unit_list
+
+    def add_unit(self, item):
```

Comment:
Would it be easier for this class to have *only* str representations OR *only* astropy.u things?  That would make it easier so that you don't have to ck each time what you have in here.